ingest/src/main/java/com/connexta/ingest/config/S3StorageConfiguration.java

```diff
@@ -42,9 +49,31 @@
   @Value("${aws.s3.bucket.quarantine}")
   private String s3BucketQuarantine;
 
-  @PostConstruct
-  public void initialize() throws IOException {
-    s3AccessKey = FileUtils.readFileToString(new File(awsAccessKeyFile), StandardCharsets.UTF_8);
-    s3SecretKey = FileUtils.readFileToString(new File(awsSecretKeyFile), StandardCharsets.UTF_8);
+  @Bean
+  public S3Client s3Client() {
+    AwsCredentials credentials = AwsBasicCredentials.create(s3AccessKey, s3SecretKey);
+
+    S3Client s3Client =
+        S3Client.builder()
+            .endpointOverride(URI.create(s3Endpoint))
+            .region(Region.of(s3Region))
+            .credentialsProvider(StaticCredentialsProvider.create(credentials))
+            .build();
+    LOGGER.info("S3 Client has been initialized");
+    return s3Client;
+  }
+
+  @Bean
+  public String s3BucketQuarantine() {
+    return s3BucketQuarantine;
+  }
+
+  public void S3StorageConfiguration() {
+    try {
+      s3AccessKey = FileUtils.readFileToString(new File(awsAccessKeyFile), StandardCharsets.UTF_8);
+      s3SecretKey = FileUtils.readFileToString(new File(awsSecretKeyFile), StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      LOGGER.error("Could not read key files.", e);
```

Comment:
✏️ Should we not catch the exception so that the app fails to start?

Reply:
Yeah that's a good point, it would fail to initialize because it's missing the key values anyway.